$OpenBSD: patch-src_gmtk_media_player_h,v 1.3 2012/06/22 15:08:45 dcoppa Exp $

Fix a subtle segmentation fault at program exit time:
gmtk_media_player_dispose() would be called multiple times, leading
to a double free from gdk_color_free()
(upstream svn revision r143)

--- src/gmtk_media_player.h.orig	Fri Jun 22 16:48:06 2012
+++ src/gmtk_media_player.h	Fri Jun 22 16:50:31 2012
@@ -385,6 +385,8 @@ struct _GmtkMediaPlayer {
     GmtkMediaPlayerMediaState restart_state;
 
     GdkColor *default_background;
+
+    gboolean disposed;
 };
 
 struct _GmtkMediaPlayerClass {
